Doc/tools/check-warnings.py

```diff
@@ -3,11 +3,13 @@
 Check the output of running Sphinx in nit-picky mode (missing references).
 """
 import argparse
```

Comment:
Line 245 uses ``list[str] | None``

```suggestion
from __future__ import annotations

import argparse
```

Reply:
Since this was originally only meant to run on the CI and is (currently) only on 3.13+ (with backport only theoretically possible to 3.11 at this point), I hadn't added it, but just in case people are running it locally on security-only versions, doesn't hurt to add, I guess.